clang/test/SemaSYCL/sycl-device-intel-max-global-work-dim-template.cpp

```diff
@@ -2,6 +2,28 @@
 
 // Test that checkes template parameter support for 'max_global_work_dim' attribute on sycl device.
 
+// Test that checks wrong function template instantiation and ensures that the type
+// is checked properly when instantiating from the template definition.
+template <typename Ty>
+// expected-error@+1{{'max_global_work_dim' attribute requires an integer constant}}
+[[intel::max_global_work_dim(Ty{})]] void func() {}
+
+struct S {};
+void var() {
+  //expected-note@+1{{in instantiation of function template specialization 'func<S>' requested here}}
+  func<S>();
```

Comment:
Another interesting test would be: `func<float>();` to see if we get an exciting implicit conversion to int there or not. Also, a test like `func<int>();`, which should be accepted.

Reply:
>>Another interesting test would be: func<float>(); to see if we get an exciting implicit conversion to int there or not.

Test is added. We get an implicit conversion to int for all attributes here.

>>Also, a test like func<int>();, which should be accepted.

Test is added.
[intel::max_global_work_dim)]] -- only accept this test
[[intel::num_simd_work_items()]] -- does not accept this test 
[[intel::reqd_sub_group_size()]] -- does not accept this test

Reply:
Is it expected that we get an implicit conversion for `func<float>()` or was that a surprise (if a surprise, maybe open an issue for it)?

Also, is the inconsistency between the other attributes with `func<int>()` expected? I would assume we'd want all of these or none of these to behave the same way. If it's unexpected, maybe open another issue?

Reply:
I have created https://github.com/intel/llvm/issues/3100 for this issue and assigned to me. I will investigate this behavior for all of these in a separate PR. Thanks @AaronBallman for suggesting about this tests.